### ISSUE TITLE
Style/ConstantVisibility does not work on non-trivial class bodies

### ISSUE THREAD

**Issue by buehmann** [rubocop-hq/rubocop#7242]
This is the issue I already mentioned [here](https://github.com/rubocop-hq/rubocop/pull/7239#issuecomment-515883573). I will add a PR fixing it shortly.

`Style/ConstantVisibility` does not do what it says on the cover.

## Steps to reproduce the problem

Enable the cop in your .rubocop.yml
```yaml
Style/ConstantVisibility:
  Enabled: true
```
Run rubocop on the following example taken straight from the [documentation](https://github.com/rubocop-hq/rubocop/blob/f51c53c0be3c16b47407c947d85beb3db21e86ac/lib/rubocop/cop/style/constant_visibility.rb#L6-L27):
```ruby
# frozen_string_literal: true

# This cop checks that constants defined in classes and modules have
# an explicit visibility declaration. By default, Ruby makes all class-
# and module constants public, which litters the public API of the
# class or module. Explicitly declaring a visibility makes intent more
# clear, and prevents outside actors from touching private state.
#
# @example

# bad
class Foo
  BAR = 42
  BAZ = 43
end

# good
class Foo
  BAR = 42
  private_constant :BAR

  BAZ = 43
  public_constant :BAZ
end
```

## Actual behavior

```
Inspecting 1 file
.

1 file inspected, no offenses detected
```

## Expected behavior

```
Inspecting 1 file
C

Offenses:

constant_visibility.rb:13:3: C: Style/ConstantVisibility: Explicitly make BAR public or private using either #public_constant or #private_constant.
  BAR = 42
  ^^^^^^^^
constant_visibility.rb:14:3: C: Style/ConstantVisibility: Explicitly make BAZ public or private using either #public_constant or #private_constant.
  BAZ = 43
  ^^^^^^^^

1 file inspected, 2 offenses detected
```

## RuboCop version

master (f51c53c0be3c16b47407c947d85beb3db21e86ac)
